### DISCUSSION TITLE
Broken with Sphinx 7.1.0 on Windows

### DISCUSSION THREAD

**Discussion by henryiii** [pradyunsg/furo#685] [Potential Issue] [answered]
The Sphinx 7.1.0 release yesterday breaks Furo on Windows; `css_tag(css)` is causing a file to be attempted to be opened with `?digest=e6660623a769aa55fea372102b9bf3151b292993` added to it, which of course breaks.

```pytb
  File "D:\a\scikit-build-core\scikit-build-core\.nox\docs\Lib\site-packages\furo\theme\furo\base.html", line 62, in block 'regular_styles'
    {{ css_tag(css) }}
  File "D:\a\scikit-build-core\scikit-build-core\.nox\docs\Lib\site-packages\jinja2\sandbox.py", line 393, in call
    return __context.call(__obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\a\scikit-build-core\scikit-build-core\.nox\docs\Lib\site-packages\sphinx\builders\html\__init__.py", line 1202, in css_tag
    if checksum := _file_checksum(app.outdir, css.filename):
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\a\scikit-build-core\scikit-build-core\.nox\docs\Lib\site-packages\sphinx\builders\html\__init__.py", line 1254, in _file_checksum
    with open(path.join(outdir, filename), encoding='utf-8') as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 22] Invalid argument: 'D:\\a\\scikit-build-core\\scikit-build-core\\docs\\_build\\html\\_static/styles/furo.css?digest=e6660623a769aa55fea372102b9bf3151b292993'
```

Seen in https://github.com/scikit-build/scikit-build-core/pull/440 and https://github.com/scikit-build/scikit-build-core/pull/441. I opened an issue with sphinx, https://github.com/sphinx-doc/sphinx/issues/11518, but this was considered an issue with Furo and not something Sphinx would fix. I think it's this is the unsupported code:

https://github.com/pradyunsg/furo/blob/f8db95bed5d1c57b4ea6bf536c65560fa9ae33a9/src/furo/__init__.py#L149-L153

**Comment by AA-Turner** [comment]
Furo could gate the asset hashes code on if the Sphinx version is 7.0 or earlier, which I believe would solve the problem?

A

**Reply by henryiii** [reply]
Don’t you mean 7.1? But that’s what I was thinking might be an easy fix. Not that I know much about it, though.

**Reply by pradyunsg** [reply]
@AA-Turner The relevant change was added in Sphinx 7.1 -- https://github.com/sphinx-doc/sphinx/pull/11415 -- so I've used 7.1 as the constraint.

Please do let me know if there's something missing here!

**Reply by AA-Turner** [reply]
That's right! I was a little unclear, I intended to say that Furo's own logic would only be applied if the Sphinx version was 7.0.* or earlier -- the opposite of which is disabling if greater than or equal to 7.1!

A

**Comment by pradyunsg** [comment, marked as answer]
Was definitely an issue!

Fixed in https://pradyunsg.me/furo/changelog/#vigilant-volt (https://github.com/pradyunsg/furo/releases/tag/2023.07.26, https://github.com/pradyunsg/furo/commit/a92dd0c2bfa398825e239171925a81bdc428e90a is the fix).
